context/app/routes_cells.py

```diff
@@ -66,6 +67,39 @@ def _first_n_matches(strings, substring, n):
     } for s, offset in zip(first_n, offsets)]
 
 
+def get_cluster_name_and_number(cluster_str):
+    cluster_name_arr = cluster_str.split('-')
+    cluster_number = cluster_name_arr.pop()
+    cluster_name = '-'.join(cluster_name_arr)
+    return (cluster_name, cluster_number)
+
+
+def get_cluster_cells(cells, gene, min_gene_expression):
+    cluster_cells = []
+    for cell in cells:
+        for cluster in cell['clusters']:
+            cluster_name, cluster_number = get_cluster_name_and_number(cluster)
+            cluster_cells.append({'modality': cell['modality'], 'cluster_name': cluster_name,
+                                  'cluster_number': cluster_number,
+                                  'meets_minimum_gene_expression':
+                                  cell['values'][gene] >= min_gene_expression})
+    return cluster_cells
+
+
+def get_matched_cell_counts_per_cluster(cells):
+    grouper = itemgetter("cluster_name", "cluster_number", 'modality')
+    clusters = {}
+    for key, grp in groupby(sorted(cells, key=grouper), grouper):
+        grp_list = list(grp)
+        cluster = dict(zip(["cluster_name", "cluster_number", 'modality'], key))
+        if not cluster['cluster_name'] in clusters:
+            clusters[cluster['cluster_name']] = []
+        cluster['matched'] = sum(item["meets_minimum_gene_expression"] for item in grp_list)
+        cluster['unmatched'] = len(grp_list) - cluster['matched']
```

Comment:
```suggestion
        cluster.update({
            'matched': sum(item["meets_minimum_gene_expression"] for item in grp_list)
            'unmatched': len(grp_list) - cluster['matched']
        })
```
Where possible, I like to have bigger data structures / comprehensions, instead of individual assignments. Easier to get a sense of what the big picture is.